src/internal/race-to.mjs

```diff
@@ -0,0 +1,23 @@
+export default function raceTo (predicate, notFoundValue, promises) {
+  return new Promise(resolve => {
+    let promiseCount = 0
+    let settledPromiseCount = 0
+    let resolved = false
+    for (const promise of promises) {
+      promise
+        .then(result => {
+          if (predicate(result)) {
+            resolve(result)
+            resolved = true
+          }
+        })
+        .finally(() => {
+          settledPromiseCount++
+          if (!resolved && settledPromiseCount === promiseCount) {
+            resolve(notFoundValue)
+          }
+        })
+      promiseCount++
```

Comment:
You have the number of promises already. Why counting them ?

Reply:
I have an iterable of promises, so I am counting how many promises I got in the iterable, and then I am also counting how many have resolved in one way or another.

Reply:
oh yes, because it is an iterable, not an array.
Maybe it would be more practical convert the iterable to array though. (see comment on promise.race)

Reply:
Again this is in a pretty tight loop. If I don't have to make an array, I won't. Inside tight loops are where iterators are most helpful.